src/main/java/com/adobe/abp/regola/rules/RangeRule.java

```diff
@@ -0,0 +1,238 @@
+/*
+ *  Copyright 2025 Adobe. All rights reserved.
+ *  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License. You may obtain a copy
+ *  of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software distributed under
+ *  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *  OF ANY KIND, either express or implied. See the License for the specific language
+ *  governing permissions and limitations under the License
+ */
+
+package com.adobe.abp.regola.rules;
+
+import com.adobe.abp.regola.facts.FactsResolver;
+import com.adobe.abp.regola.results.Result;
+import com.adobe.abp.regola.results.RuleResult;
+import com.adobe.abp.regola.results.ValuesRuleResult;
+import com.adobe.abp.regola.utils.futures.FutureUtils;
+import jdk.jfr.Experimental;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * Experimental rule for evaluating facts against a range of values.
+ * Supports basic range operations for comparable types.
+ *
+ * The rule supports the following operators:
+ * <ul>
+ * <li>{@link Operator#BETWEEN}</li>
+ * <li>{@link Operator#IN}</li>
+ * <li>{@link Operator#IS_BEFORE}</li>
+ * <li>{@link Operator#IS_AFTER}</li>
+ * </ul>
+ *
+ * @param <T> type of the fact (must extend Comparable)
+ */
+@Experimental
+public class RangeRule<T extends Comparable<T>> extends OperatorBasedRule {
```

Comment:
No need for an update to the README as this is experimental for now, but do have unit tests please.